### PR TITLE
(PUP-8305) Enable git-based acceptance testing

### PR DESCRIPTION
**Pull request by melissa**
This commit fixes a few bugs that surfaced as a part of the effort to
get git-based acceptance tests working

### CODE REVIEW

**Comment by melissa** [comment]
```
ClassMixedWithDSLStructure
  #select_hosts
    selects hosts when a passed block returns true (FAILED - 1)

Failures:

  1) ClassMixedWithDSLStructure#select_hosts selects hosts when a passed block returns true
     Failure/Error: expect( selected_hosts ).to be == [ host1 ]

       expected: == [{"platform"=>"solaris1"}]
            got:    []
     # ./spec/beaker/dsl/structure_spec.rb:379:in `block (3 levels) in <top (required)>'

Finished in 0.01893 seconds (files took 0.76486 seconds to load)
1 example, 1 failure
```

```
362     it 'selects hosts when a passed block returns true' do
363       host1 = {'platform' => 'solaris1'}
364       host2 = {'platform' => 'solaris2'}
365       host3 = {'platform' => 'windows'}
366       ret1 = double('result1')
367       allow( ret1 ).to receive( :stdout ).and_return(':global')
368       ret2 = double('result2')
369       allow( ret2 ).to receive( :stdout ).and_return('a_zone')
370       hosts = [ host1, host2, host3 ]
371       expect( subject ).to receive( :hosts ).and_return( hosts )
372
373       expect( subject ).to receive( :on ).with( host1, '/sbin/zonename' ).once.and_return( ret1 )
374       expect( subject ).to receive( :on ).with( host2, '/sbin/zonename' ).once.and_return( ret2 )
375
376       selected_hosts = subject.select_hosts 'platform' => 'solaris' do |host|
377                              subject.on(host, '/sbin/zonename').stdout =~ /:global/
378       end
379       expect( selected_hosts ).to be == [ host1 ]
380     end
```

This is the only failure. It's returning an empty value because `solaris` != `solaris1`. I recognize that there is some functionality change that's happening with this, but it seems like this test doesn't mean to be testing that. When I try to preserve functionality, there are a bunch of additional test failures. So is this test not testing the correct thing, or am I misunderstanding the purpose of `inspect_host`?

File: lib/beaker/dsl/structure.rb [hunk 1]
```diff
@@ -350,7 +350,7 @@ def inspect_host(host, property, one_or_more_values)
           true_false = false
           case value
           when String
-            true_false = host[property.to_s].include? value
+            true_false = Array(host[property.to_s]).include? value
```
[the remark this answers is not in the document]
Comment (melissa) [reply lib/beaker/dsl/structure.rb]: I'm not able to reproduce the failure that I was trying to fix with this change, so I'm just going to back it out

**Comment by melissa** [comment]
Tests pass!

**Comment by melissa** [comment]
Nope, I am still seeing that failure
```
NoMethodError: undefined method `include?' for :git:Symbol
/Users/Melissa/beaker/lib/beaker/dsl/structure.rb:353
```


The test that's failing is failing on this line:
```
confine :except, :type => 'pe'
```

It's checking type, so `host[property.to_s]`, aka `host['type']` is returning `:git`. It's trying to run `.include?` on a symbol, which isn't a valid method. I think a better solution would be casting this to a string rather than an array. I'll amend this pr momentarily.

**Comment by Dakta** [comment]
> `host[property.to_s]`, aka `host['type']` is returning `:git`

Host property `type` is supposed to be a String: [nowhere in the docs is Symbol representation demonstrated or specified](https://github.com/puppetlabs/beaker/search?l=Markdown&p=2&q=type). Where is it being set to Symbol, your hosts file? This is a misconfiguration.

**Comment by melissa** [comment]
ooh, funny. That was changed a few years ago, but since what I've been working on has been broken for forever, no one noticed. Thanks! I've again removed that change

**Comment by melissa** [comment]
Is there a good way to rekick these tests? I'm not seeing the failure reported, and I'd like to get this in soon. I was hoping to catch the release that just went out, but since we missed that, when can we get this out?

**Comment by melissa** [comment]
@Dakta @kevpl ping!

**Pull request merged**